### PR TITLE
DPC-592: Fix rails app bug from password update

### PR DESCRIPTION
**Pull request by Sun-Mountain**
### Fixes [DPC-592](https://jira.cms.gov/browse/DPC-592)

Fixing bugs introduced to master in two previous PRs.
- Broken Javascript design transitions (nav bar, drop down from org select)
- Broken spec tests

### Proposed Changes

- This [PR](https://github.com/CMSgov/dpc-app/pull/966) fixes the bug that didn't catch the broken spec tests
- Fix spec tests to organization search from [DPC-547](https://github.com/CMSgov/dpc-app/pull/939)
- Reverted to an earlier PR and redid the Javascript that broke the design transitions

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### CODE REVIEW

File: dpc-web/app/controllers/client_tokens_controller.rb [hunk 1]
```diff
@@ -11,9 +11,11 @@ def new
 
   def create
     @organization = current_user.organizations.find(params[:organization_id])
+    return render_error('Must have a label.') if missing_params
 
     reg_org = @organization.registered_organization
     manager = ClientTokenManager.new(registered_organization: reg_org)
+
     if missing_params
```
Comment (dhgreene) [inline dpc-web/app/controllers/client_tokens_controller.rb:19]: Having already tested for a missing parameter, should we remove the if/elsif here?

File: dpc-web/app/controllers/internal/organizations_controller.rb [hunk 2]
```diff
@@ -87,26 +89,33 @@ def add_or_delete
       @organization = Organization.find(params[:organization_id])
       @user = user_identify
 
-      add_delete(params)
+      if params[:_method] == 'add'
+        add_user_to_org
+      elsif params[:_method] == 'delete'
+        delete_user_from_org
+      else
+        return redirect_to internal_organization_path(@organization)
```
Comment (dhgreene) [inline dpc-web/app/controllers/internal/organizations_controller.rb:97]:
```suggestion
        redirect_to internal_organization_path(@organization)
```

File: dpc-web/app/controllers/internal/users_controller.rb [hunk 3]
```diff
@@ -28,6 +28,7 @@ def update
         flash[:notice] = 'User successfully updated.'
         redirect_to internal_user_url(@user)
       else
+        @organizations = Organization.all
```
Comment (dhgreene) [inline dpc-web/app/controllers/internal/users_controller.rb:31]: Just double-checking: I don't see `@organizations used directly by the view or partial.  What breaks if we don't have this line?

File: dpc-web/config/initializers/devise-security.rb [hunk 4]
```diff
@@ -9,15 +9,15 @@
...
+  # How many passwords to keep in archive
   config.password_archiving_count = 12
-  
+
   # Deny old passwords (true, false, number_of_old_passwords_to_check)
   # Examples:
   # config.deny_old_passwords = false # allow old passwords
   # config.deny_old_passwords = true # will deny all the old passwords
-  config.deny_old_passwords = 12 # will deny new passwords that matches with the last 12 passwords
+  config.deny_old_passwords = 12 # will deny new passwords that matches with the last 3 passwords
```
Comment (dhgreene) [inline dpc-web/config/initializers/devise-security.rb:20]: The comment no longer seems true . . .

File: dpc-web/app/controllers/internal/users_controller.rb (hunk 3 above)
Comment (Sun-Mountain) [reply dpc-web/app/controllers/internal/users_controller.rb:31]: It's in the partial `_org_search.html.erb` with `There are <%= @organizations.count %> organizations available.`

If I delete the line in the controller, I can replace it with `Organization.all`.

**Pull request merged**